stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -311,21 +341,28 @@ internal class PaymentSheetViewModel internal constructor(
         viewModelScope.launch {
             val result = runCatching {
                 withContext(workContext) {
-                    paymentFlowResultProcessor.processPaymentIntent(paymentFlowResult)
+                    when (args.clientSecret) {
```

Comment:
Let's have a DefaultFlowResultProcessor for setup intent and payment intent, then we don't need to know the difference here.

Reply:
let's talk this through

Reply:
DefaultPaymentFlowResultProcessor isn't in paymentsheet, so it might be better to create your own DefaultSetupIntentFlowResultProcessor and DefaultPaymentIntentFlowResultProcessor that implement a IntentFlowResultProcessor.